File: Selenium/Project/Activity5.java
package Projects_HRM;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity5 {
    static WebDriver driver;

    @BeforeClass
    public static void setup() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        //Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        driver.get("http://alchemy.hguy.co/orangehrm");
    }

    @Test
    public void verifyInfo() {
        WebElement username = driver.findElement(By.xpath("//input[@id = 'txtUsername']"));
        WebElement password = driver.findElement(By.xpath("//input[@id = 'txtPassword']"));

        username.sendKeys("orange");
        password.sendKeys("orangepassword123");
        driver.findElement(By.xpath("//input[@id = 'btnLogin']")).click();
        WebElement myInfo = driver.findElement(By.xpath("//*[@id='menu_pim_viewMyDetails']"));
        myInfo.click();
        driver.findElement(By.xpath("//input[@value = 'Edit']")).click();
        WebElement firstname = driver.findElement(By.xpath("//input[@title= 'First Name']"));
        WebElement lastName = driver.findElement(By.xpath("//input[@title= 'Last Name']"));

        firstname.clear();
        firstname.sendKeys("First Name");
        lastName.clear();
        lastName.sendKeys("Last Name");
        driver.findElement(By.xpath("//*[@id= 'personal_optGender_1']")).click();
        Select select = new Select(driver.findElement(By.xpath("//select[@id= 'personal_cmbNation']")));

        select.selectByVisibleText("Indian");
        WebElement dob = driver.findElement(By.xpath("//input[@id= 'personal_DOB']"));

        dob.sendKeys("1995-03-21");
        driver.findElement(By.xpath("//input[@id= 'btnSave']")).click();
    }

    @AfterClass
    public static void afterMethod() {
        //Close the browser
        driver.quit();
    }
}

File: Selenium/Project/Activity5.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.ui import Select

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login")
    username = driver.find_element(By.XPATH,"//input[@id = 'txtUsername']")
    password = driver.find_element(By.XPATH,"//input[@id = 'txtPassword']")
    username.send_keys("orange")
    password.send_keys("orangepassword123")
    driver.find_element(By.XPATH,"//input[@id = 'btnLogin']").click()
    myInfo = driver.find_element(By.XPATH,"//*[@id = 'menu_pim_viewMyDetails']")
    myInfo.click()
    driver.find_element(By.XPATH,"//input[@value = 'Edit']").click()
    firstName = driver.find_element(By.XPATH,"//input[@title= 'First Name']")
    lastName = driver.find_element(By.XPATH,"//input[@title= 'Last Name']")
    firstName.clear()
    firstName.send_keys("Abhi")
    lastName.clear()
    lastName.send_keys("ghghg")
    driver.find_element(By.XPATH,"//*[@id= 'personal_optGender_1']").click()
    select = Select(driver.find_element(By.XPATH, "//select[@id= 'personal_cmbNation']"))
    select.select_by_visible_text("Indian")
    dob = driver.find_element(By.XPATH,"//input[@id= 'personal_DOB']")
    dob.send_keys("1993-09-20")
    driver.find_element(By.XPATH,"//input[@id= 'btnSave']").click()
    driver.quit()